### PR TITLE
Feature/query wizard

### PR DESCRIPTION
**Pull request by ca1yps0**

### CODE REVIEW

**Comment by nightsh** [comment]
- steps line is not aligned properly (1-2 px to the right should do)
- "loading" bits are not set on wizard step change (hence a slower "loading" feedback)
- step titles look like anchors, they shouldn't
- settings: images don't have correct paths
- settings drop downs styling

**Review by victor-nitu — CHANGES_REQUESTED** [review]
- [x] you should receive an error message if you didn't select anything
  - [x] if nothing was selected in step 1, allow step 2 but don't allow step 3 if nothign was selected
- [x] if you select a company, then you go back and change countries, when you go back to actors the old selection is still there (make it disappear if you change any previous steps)
- [x] steps line is not aligned properly (1-2 px to the right should do)
- [x] "loading" bits are not set on wizard step change (hence a slower "loading" feedback)
- [x] step titles look like anchors, they shouldn't
- [x] settings: images don't have correct paths
- [x] settings drop downs styling

**Review by victor-nitu — APPROVED** [review]

File: app/components/actor-select-guru.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import EmberSelectGuru from 'ember-select-guru/components/ember-select-guru';
+import { run } from '@ember/runloop';
+import { observer } from '@ember/object';
+import { get } from '@ember/object';
+
+export default EmberSelectGuru.extend({
+  // checkActors: false,
+  checkboxIsSelected: false,
+  /*selectAllOptions: observer('checkboxIsSelected', function() {
```
Comment (victor-nitu) [inline app/components/actor-select-guru.js]: dead code

File: app/components/actor-select-guru.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import EmberSelectGuru from 'ember-select-guru/components/ember-select-guru';
+import { run } from '@ember/runloop';
+import { observer } from '@ember/object';
+import { get } from '@ember/object';
+
+export default EmberSelectGuru.extend({
+  // checkActors: false,
```
Comment (victor-nitu) [inline app/components/actor-select-guru.js]: dead code

File: app/components/actor-select-guru.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+      _.each(selected, (option) => {
+        if (_.indexOf(self.get('countries'), option.country) >= 0) {
+          this.get('_value').pushObject(option);
+        }
+        this.set('value', _.cloneDeep(this.get('_value')));
+      });
+      let onSelect = get(this, 'onSelect');
+      onSelect(this.get('_value'));
+      // this.attrs.onSelect(this.get('_value'));
```
Comment (victor-nitu) [inline app/components/actor-select-guru.js]: more dead code

**Review by victor-nitu — CHANGES_REQUESTED** [review]
Not going to go through all the diff, please remove all dead code before anything.

**Comment by zufanka** [comment]
Great job guys. @ca1yps0 @nightsh @chutko 
There are still a couple of things to fix:

**BUGS**
- [ ] if you don't select a CPV, the "Settings" step is empty
- [ ] if you go to "Create a network" from the network view (back to the query builder), the previous countries are actually still selected, but you don't see it in the first step (if you go a step further, the flags appear). Everything should be deselected if you do this
- [ ] when I press "Launch the network", a loader should appear
- [ ] flags missing here
image
- [ ] This errormessage appears when server returns 500. This should NOT be the case as it's NOT the problem. Can we change it to "This should not be happening. Please send us an email at tech@tenders.exposed if it is." or something like that?
image


**STYLING**
- [ ] this selection goes into the footer. Please give the element the same height all the steps in the the whole wizard
image
- [ ] there should be more space between these
image
- [ ] there should be definitely more space here
image
- [ ] the "Node types" and "edge types" as well as the "name of the network" are a bit small and difficult to read. @chutko can we make them pop out more?
image
- [ ] can we have these on a transparent background? @chutko 
image
- [ ] There should be some indicator that the user chose actors here (no countries chosen now)
image

**Comment by victor-nitu** [comment]
@zufanka I will merge and copy your message into a new issue - we will surely rebranch after merging but this is getting again and again out of sync with develop and it's becoming a hassle to maintain it open & mergeable for so long.

**Pull request merged**